src/main/java/trycb/service/Hotel.java

```diff
@@ -42,43 +43,27 @@ public Hotel(Bucket bucket) {
     /**
      * Search for a hotel in a particular location.
      */
-    public Result<List<Map<String, Object>>> findHotels(final Cluster cluster, final String location, final String description) {
+    public Result<List<Map<String, Object>>> findHotels(final Cluster cluster, final String location,
+            final String description) {
         ConjunctionQuery fts = SearchQuery.conjuncts(SearchQuery.term("hotel").field("type"));
 
         if (location != null && !location.isEmpty() && !"*".equals(location)) {
-            fts.and(SearchQuery.disjuncts(
-                        SearchQuery.matchPhrase(location).field("country"),
-                        SearchQuery.matchPhrase(location).field("city"),
-                        SearchQuery.matchPhrase(location).field("state"),
-                        SearchQuery.matchPhrase(location).field("address")
-                ));
+            fts.and(SearchQuery.disjuncts(SearchQuery.matchPhrase(location).field("country"),
+                    SearchQuery.matchPhrase(location).field("city"), SearchQuery.matchPhrase(location).field("state"),
+                    SearchQuery.matchPhrase(location).field("address")));
         }
 
         if (description != null && !description.isEmpty() && !"*".equals(description)) {
-            fts.and(
-                SearchQuery.disjuncts(
-                        SearchQuery.matchPhrase(description).field("description"),
-                        SearchQuery.matchPhrase(description).field("name")
-                ));
+            fts.and(SearchQuery.disjuncts(SearchQuery.matchPhrase(description).field("description"),
+                    SearchQuery.matchPhrase(description).field("name")));
         }
 
         logQuery(fts.export().toString());
         SearchOptions opts = SearchOptions.searchOptions().limit(100);
-        SearchResult result = cluster.searchQuery("hotels", fts, opts);
-
-        //prepare the context to send to the app
-        String ftsContext;
-        try {
-            ftsContext = JacksonTransformers.MAPPER.writerWithDefaultPrettyPrinter().
-                    writeValueAsString(fts.export());
-        } catch (JsonProcessingException e) {
-            ftsContext = fts.export().toString();
-        }
-        String subdocContext = "        Optional<LookupInResult> lookup = bucket.defaultCollection().lookupIn(row.id(),\n" +
-                "                Arrays.asList(get(\"country\"), get(\"city\"), get(\"state\"), get(\"address\"),\n" +
-                "                        get(\"name\"), get(\"description\")));";
```

Comment:
don't quite understand this, but looks like you've got rid of some useless, ugly code, so yay!

Reply:
I think it was trying to show the code that gets executed in the narration via a _string_ representation, but it just looks weird and confusing.

Plus a user can just see the code in the app, and it would be a pain to update this if the code changes.

This was what it was trying to show I think (but see already the code changed and the string hadn't been updated): https://github.com/couchbaselabs/try-cb-java/blob/cc831f4d9609d1dd3bd8e62ff9b819e139e788db/src/main/java/trycb/service/Hotel.java#L110-L114

Reply:
Wow, yes, that seems ill-advised, I wonder if there was a specific request at the time to focus on that feature... anyway, good call to remove it :D